drivers/serial/uart_stm32.c

```diff
@@ -1377,6 +1377,9 @@ static int uart_stm32_async_tx_abort(const struct device *dev)
 		data->dma_tx.counter = tx_buffer_length - stat.pending_length;
 	}
 
+#if CONFIG_DT_HAS_ST_STM32U5_DMA_ENABLED
```

Comment:
`CONFIG_DT_HAS_<FOO>_ENABLED` are restricted symbols, it's not expected to be used in code.
Use
```suggestion
#if CONFIG_DMA_STM32U5
```
or
```
#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32_dma_u5)
```

Reply:
using `#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32u5_dma)`